File: CS/E1887/PersistentClasses.cs
using DevExpress.Xpo;
namespace DXSample.BO {
    public class Order : XPObject {
        public Order(Session session) : base(session) { }

        private string fName;
        public string Name {
            get { return fName; }
            set { SetPropertyValue<string>("Name", ref fName, value); }
        }

        private decimal fUnitPrice;
        public decimal UnitPrice {
            get { return fUnitPrice; }
            set { SetPropertyValue<decimal>("UnitPrice", ref fUnitPrice, value); }
        }

        private int fQuantity;
        public int Quantity {
            get { return fQuantity; }
            set { SetPropertyValue<int>("Quantity", ref fQuantity, value); }
        }

        private Order fParent;
        [Association("Order-SubOrders")]
        public Order Parent {
            get { return fParent; }
            set { SetPropertyValue<Order>("Parent", ref fParent, value); }
        }

        [Association("Order-SubOrders", typeof(Order))]
        public XPCollection<Order> SubOrders { get { return GetCollection<Order>("SubOrders"); } }
    }
}

File: VB/E1887/PersistentClasses.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Namespace DXSample.BO
	Public Class Order
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private fName As String
		Public Property Name() As String
			Get
				Return fName
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Name", fName, value)
			End Set
		End Property

		Private fUnitPrice As Decimal
		Public Property UnitPrice() As Decimal
			Get
				Return fUnitPrice
			End Get
			Set(ByVal value As Decimal)
				SetPropertyValue(Of Decimal)("UnitPrice", fUnitPrice, value)
			End Set
		End Property

		Private fQuantity As Integer
		Public Property Quantity() As Integer
			Get
				Return fQuantity
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue(Of Integer)("Quantity", fQuantity, value)
			End Set
		End Property

		Private fParent As Order
		<Association("Order-SubOrders")> _
		Public Property Parent() As Order
			Get
				Return fParent
			End Get
			Set(ByVal value As Order)
				SetPropertyValue(Of Order)("Parent", fParent, value)
			End Set
		End Property

		<Association("Order-SubOrders", GetType(Order))> _
		Public ReadOnly Property SubOrders() As XPCollection(Of Order)
			Get
				Return GetCollection(Of Order)("SubOrders")
			End Get
		End Property
	End Class
End Namespace